### ISSUE TITLE
Allow columns to be initially hidden [bounty available]. 

### ISSUE THREAD

**Issue by neilgilmour** [neilgilmour/inertiajs-tables-laravel-query-builder#1]
The feature is for allowing columns to be hidden initially in the front end by passing `false` to the `add Column()` method in the controller. I've added the back end code for this, but on the front end I'm having trouble; the `queryBuilderProps.columns[columnKey].enabled` comes through as false. However when a column is toggled, `queryBuilderData.columns` becomes an array of columns that are toggled *on*, but my Vue skills aren't enough to follow through methods in the mixin to make this work properly.

I need this fairly quickly, so I'm offering £150 (c. $208) for the first PR/patch to this repo that makes this work.

**Comment by luckyboy27** [comment]
You can use `temp` data for it.

**Comment by AmirJavaidSwe** [comment]
Hi @neilgilmour 

I can fix this issue for you, if it is still available.

Shall I do?

**Comment by neilgilmour** [comment]
> You can use `temp` data for it.

What would that look like?

**Comment by arsangamal** [comment]
@neilgilmour in the js/Components/Table.vue there is a method 
```js
changeColumnStatus(column, status) {
      this.queryBuilderData.columns[column].enabled = status;
},
```
try to use on every update you make to the query builder
```js
changeColumnStatus('column_name', false)
```

or the more professional solution
create a variable that holds the pre-hidden columns
```js
<template #body>
      <tr v-for="user in users.data" :key="user.id">
        <td>{{ user.name }}</td>
        <td v-show="hiddenColumns.includes('email') && showColumn('email')">{{ user.email }}</td>
        <td v-show="showColumn('language_code')">{{ user.language_code }}</td>
      </tr>
</template>
```

```js
export default {
  mixins: [InteractsWithQueryBuilder],
  
  data:{
     return {
        hiddenColumns:[ 'email' ],
     }|
  },
  
  components: {
    Table: Tailwind2.Table
  },

  props: {
    users: Object
  }
};

```
in this case the email will be hidden

**Comment by neilgilmour** [comment]
Thanks @arsangamal, if I do the second two blocks of code, then only the fields that are true by default AND in the `hidden` array are displayed. If I do `<td v-show="!hiddenColumns.includes('email') && showColumn('email')">{{ user.email }}</td>` then I get the correct columns displayed initially, but the toggle shows that hidden columns are visible, and toggling them doesn't work, so columns in the `hidden` array never show.

**Comment by arsangamal** [comment]
@neilgilmour So you can move the hidden array and functionality inside the TableColumns.vue so you control both hidden array and toggle functionality from the same source, am i clear ?

**Comment by neilgilmour** [comment]
Thanks for everyone's help. Fixed by #2
